includes/helper-functions.php

```diff
@@ -0,0 +1,305 @@
+				isset( $temp_arr[ $counter ]['font_url'] ) && '' !== $temp_arr[ $counter ]['font_url'] &&
+				isset( $temp_arr[ $counter ]['font_file'] ) && '' !== $temp_arr[ $counter ]['font_file'] &&
+				isset( $temp_arr[ $counter ]['font_weight'] ) && '' !== $temp_arr[ $counter ]['font_weight']
+			) {
+				$counter = $counter + 1;
+			}
+
+			$consider_counter = false;
+
+			if ( strpos( $key, 'display' ) !== false ) {
+				$font_data['font_display'] = $value;
+				$consider_counter          = true;
+			}
+			if ( strpos( $key, 'fallback' ) !== false ) {
+				$font_data['font_fallback'] = $value;
+				$consider_counter           = true;
+			}
+			if ( strpos( $key, 'weight' ) !== false ) {
+				$temp_arr[ $counter ]['font_weight'] = $value;
+				$consider_counter                    = true;
+			}
+
+			if ( strpos( $key, 'font_woff_2' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+			if ( strpos( $key, 'font_woff-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+			if ( strpos( $key, 'font_ttf-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+			if ( strpos( $key, 'font_eot-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+			if ( strpos( $key, 'font_svg-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+			if ( strpos( $key, 'font_otf-' ) !== false && $value ) {
+				$consider_counter                  = true;
+				$temp_arr[ $counter ]['font_url']  = esc_url( $value );
+				$temp_arr[ $counter ]['font_file'] = bcf_get_media_image_id_by_url( $value );
+			}
+
+			if ( true === $consider_counter ) {
+				$temp_arr[ $counter ]['id']         = $counter + 1;
+				$temp_arr[ $counter ]['font_style'] = 'normal';
+			}
+		}
+
+		$font_data['variations'] = $temp_arr;
+	}
+
+	return $font_data;
+}
+
+/**
+ * Sanitize data recursively, eg: font-face data.
+ *
+ * @since x.x.x
+ */
+function bcf_sanitize_text_field_recursive( $data ) {
+	if ( is_array( $data ) ) {
+		foreach ( $data as $key => $value ) {
+			$data[ $key ] = bcf_sanitize_text_field_recursive( $value );
+		}
+
+		return $data;
+	}
+	return sanitize_text_field( $data );
+}
+
+/**
+ * Get font source as per the asked type.
+ *
```

Comment:
@params are not added to the function @imnavanath

Reply:
DONE